nuxeo-webengine/nuxeo-webengine-core/src/main/java/org/nuxeo/ecm/webengine/forms/FormData.java

```diff
@@ -221,16 +226,25 @@ public Blob getFirstBlob() {
 
     protected Blob getBlob(FileItem item) {
         try {
-            InputStream in;
+            Blob blob;
             if (item.isInMemory()) {
-                in = new ByteArrayInputStream(item.get());
+                blob = Blobs.createBlob(item.get());
             } else {
-                in = item.getInputStream();
+                File file;
+                if (item instanceof DiskFileItem //
+                        && (file = ((DiskFileItem) item).getStoreLocation()) != null) {
+                    // move the file to a temporary blob we own
+                    blob = Blobs.createBlobWithExtension(null);
+                    Files.move(file.toPath(), blob.getFile().toPath(), REPLACE_EXISTING);
+                } else {
+                    // if we couldn't get to the file, use the InputStream
+                    try (InputStream in = item.getInputStream()) {
+                        blob = Blobs.createBlob(in);
+                    }
+                }
             }
-            String ctype = item.getContentType();
-            Blob blob = Blobs.createBlob(in, ctype == null ? "application/octet-stream" : ctype);
+            blob.setMimeType(defaultIfEmpty(item.getContentType(), "application/octet-stream"));
```

Comment:
Use a constant?

Reply:
Again just refactoring existing code. We should do a global pass on the whole platform to have constants...